Source/WebKit/UIProcess/Extensions/WebExtensionDynamicScripts.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+
+#include "WebExtension.h"
+#include "WebExtensionScriptInjectionResultParameters.h"
+#include "WebPageProxyIdentifier.h"
+#include <wtf/Forward.h>
+
+OBJC_CLASS WKContentWorld;
+OBJC_CLASS WKFrameInfo;
+OBJC_CLASS WKWebView;
+OBJC_CLASS _WKFrameTreeNode;
+
+namespace WebKit {
+
+namespace DynamicScripts {
+using SourcePair = std::pair<String, std::optional<URL>>;
+using StyleSheetPairs = Vector<std::optional<SourcePair>>;
+
+static const size_t mainFrameID = 0;
```

Comment:
I'd be very careful defining constants in a header. Is there some other place we can get this magic value. Also is this even the correct ID? I think 0 is a placeholder ID rather than the main frame ID.

Reply:
This is in a new namespace, but this is already defined as `WebExtensionFrameConstants::MainFrame` so this can be removed.